dev/benchmarks/macrobenchmarks/lib/src/web/gallery/gallery_recorder.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import 'package:gallery/benchmarks/gallery_automator.dart';
+
+import 'package:macrobenchmarks/src/web/recorder.dart';
+
+/// A recorder that measures frame building durations for the Gallery.
+class GalleryRecorder extends WidgetRecorder {
+  GalleryRecorder({
+    @required this.benchmarkName,
+    @required this.reporter,
+    @required this.reportError,
+    @required this.runCriterion,
+    this.testScrollsOnly = false,
+  }) : super(name: benchmarkName, warmUpFrameCount: 5);
+
+  /// The name of the gallery benchmark to be run.
+  final String benchmarkName;
+
+  /// A function that reports a given message.
+  ///
+  /// When connected to a server, sends a request to the server.
+  /// When used in `flutter run`, uses `print`.
+  final void Function(String) reporter;
+
+  /// A function that reports a given error.
+  ///
+  /// When connected to a server, sends a request to the server.
+  /// Also `print`s the error message.
+  final Future<void> Function(dynamic error, StackTrace stackTrace) reportError;
+
+  /// A function accepting the name of a demo, returning whether we should run
+  /// this demo in this benchmark.
+  final bool Function(String) runCriterion;
```

Comment:
We prefer the term "predicate" instead of "criterion".

Reply:
I'll use `runPredicate`.

Reply:
Done.